src/main/java/wromaciej/hvac_sim/solver/externals/JunctionSolver.java

```diff
@@ -0,0 +1,90 @@
+package wromaciej.hvac_sim.solver.externals;
+
+import org.springframework.stereotype.Service;
+import wromaciej.hvac_sim.solver.result.SolverResult;
+import wromaciej.hvac_sim.solver.result.SolverResultType;
+import wromaciej.hvac_sim.thermo.generals.bonds.BondDirection;
+import wromaciej.hvac_sim.thermo.generals.conservationLaw.Junction;
+import wromaciej.hvac_sim.thermo.generals.conservationLaw.ParameterWithDirection;
+import wromaciej.hvac_sim.thermo.matter.fluids.parameters.Parameter;
+import wromaciej.hvac_sim.thermo.matter.fluids.parameters.ParameterType;
+
+import java.util.ArrayList;
+import java.util.List;
+
+@Service
+public class JunctionSolver implements ExternalSolver<Junction> {
+
+
+    public List<ParameterWithDirection> getAllParameters(Junction toSolve) {
+        return toSolve.getAllParameters();
+    }
+
+    public List<ParameterWithDirection> getUndefinedParameters(Junction toSolve) {
+        List<ParameterWithDirection> undefinedParameters = new ArrayList<>();
+        for (ParameterWithDirection parameterWithDirection : getAllParameters(toSolve)) {
+            if (!parameterWithDirection.getParameter().isDefined()) undefinedParameters.add(parameterWithDirection);
+        }
+
+        return undefinedParameters;
+    }
+
+    public List<ParameterWithDirection> getDefinedParameters(Junction toSolve) {
+        List<ParameterWithDirection> definedParameters = new ArrayList<>();
+        for (ParameterWithDirection parameterWithDirection : getAllParameters(toSolve)) {
+            if (parameterWithDirection.getParameter().isDefined()) definedParameters.add(parameterWithDirection);
+        }
+        return definedParameters;
+    }
+
+    public JunctionSolver() {}
+
+    private int allParametersNumber(Junction toSolve){
```

Comment:
name should be rather `countAllParameters`. However why that is not method of Junction itself?